app/controllers/state_file/archived_intakes/mailing_address_validation_controller.rb

```diff
@@ -0,0 +1,46 @@
+require 'csv'
+
+module StateFile
+  module ArchivedIntakes
+    class MailingAddressValidationController < ArchivedIntakeController
+      before_action :check_feature_flag
+      before_action :is_request_locked
+      before_action :confirm_code_and_ssn_verification
+      def edit
+        create_state_file_access_log("issued_mailing_address_challenge")
+        @addresses = address_challenge_set
+        @form = MailingAddressValidationForm.new(addresses: @addresses, current_address: current_archived_intake.full_address)
+      end
+
+      def update
+        @form = MailingAddressValidationForm.new(mailing_address_validation_form_params, addresses: @addresses, current_address: current_archived_intake.full_address)
+        @addresses = address_challenge_set
+
+        if @form.valid?
+          create_state_file_access_log("correct_mailing_address")
+          # this should take us to the download
+          redirect_to root_path
```

Comment:
I like the pattern that we did in our ticket (which was @mrotondo's suggestion) of adding the ticket that will fix this.

Reply:
do you want to add session data for the mailing challenge to be completed?